src/main/java/com/cucumber/junit/driver/DriverManager.java

```diff
@@ -1,32 +1,50 @@
 package com.cucumber.junit.driver;
+
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
 
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 
 public class DriverManager {
 
-    private  static final String CHROME_DRIVER_PATH = "src/main/resources/chromedriver.exe";
-    private static ThreadLocal<WebDriver> webDriverThreadLocal = new ThreadLocal<>();
+    private static final String CHROME_DRIVER_PATH = "src/main/resources/chromedriver.exe";
+    private static final String GECKO_DRIVER_PATH = "src/main/resources/geckodriver.exe";
+    private static final ThreadLocal<WebDriver> webDriverThreadLocal = new ThreadLocal<>();
 
     private DriverManager() {
     }
 
-    public static void setupDriver(){
-        System.setProperty("webdriver.chrome.driver", CHROME_DRIVER_PATH);
-        WebDriver driver = new ChromeDriver();
+    public static void setupDriver() {
+
+        WebDriver driver = null;
+        String browser = System.getProperty("browser");
+
+        if (browser.equalsIgnoreCase("Chrome")) {
+            System.setProperty("webdriver.chrome.driver", CHROME_DRIVER_PATH);
+            driver = new ChromeDriver();
+        }
+        else if (browser.equalsIgnoreCase("firefox")) {
+            System.setProperty("webdriver.firefox.driver", GECKO_DRIVER_PATH);
+            driver = new FirefoxDriver();
+        }
+        else {
+            throw new IllegalStateException("This driver is not supported: " + System.getProperty("browser"));
```

Comment:
Please use the existing "browser" String variable, no need to execute getProperty() one more time

Reply:
done